packages/client/src/router/withPrivateRoute.tsx

```diff
@@ -0,0 +1,13 @@
+import { Navigate } from 'react-router-dom';
+import { RouterList } from './routerList';
+
+const checkAuth = (): boolean => {
+  //ToDo проверку авторизации вынести в контроллер, "false"- нет доступа
```

Comment:
**Можно лучше**

Остальной код не вызвал вопросов, но вот использование TODO-комментария их вызывает. Главная претензия к ним - их трудно отслеживать, отчего легко забыть реализовать требуемую функциональность. 

Советую почитать статью https://medium.com/@bpnorlander/stop-writing-code-comments-28fef5272752, а пока процитирую:

_Don’t write TODO comments, instead maybe just… do it? Most of the time these comments get forgotten about and later on might become irrelevant or wrong. When another coder sees the TODO comment later on, how do they know if this still needs to be done? Delete these comments too._

_The only time a TODO comment is ok is if you are waiting for a merge from another teammate. This is not meant to last a long time, just until you can make the fix and submit it._

_“When you feel the need to write a comment, first try to refactor the code so that any comment becomes superfluous.” — Martin Fowler_

Поэтому рекомендую сразу реализовать эту функциональность. Обрати внимание на статью https://blog.netcetera.com/how-to-create-guarded-routes-for-your-react-app-d2fe7c7b6122, хотя там описано использование гардов, вместо приватных роутов, тем не менее, в конце статьи есть прекрасная иллюстрация того, чего ожидаешь увидеть после решения задачи.

В тему приватных роутов и их реализации оставлю здесь статью
https://www.robinwieruch.de/react-router-private-routes/

Наконец, дополню про TODO-комментарии... поскольку иногда необходимость в них возникает, то разумно придерживаться стандартов оформления и использовать специальные расширения для IDE, которые упрощают работу с ними, например  ToDo Tree для VS Code

А если в TODO описана функциональность, которую не разумно реализовывать в текущей задаче, то стоит завести Issue в таск-трекере и тогда уже оставить в коде TODO с номером Issue. Тогда задача не потеряется, т.к. она хранится в таск-трекере. Не возникнет вопроса "а актуален ли вообще коммент", т.к. ты всегда можешь глянуть статус Issue. А также ты сам не потеряешь точку в коде, откуда начать, т.к. TODO-комментарий оставлен.